gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDataset.java

```diff
@@ -0,0 +1,294 @@
+   *   * if {@link #RELOCATE_DATA_FILES_KEY} is true, then origin file /path/to/file/myFile will be written to
+   *     /resolved/table/root/<partition>/myFile
+   *   * otherwise, if the resolved table root is defined (e.g. {@link #COPY_TARGET_TABLE_ROOT} is defined in the
+   *     properties), we define:
+   *     origin_table_root := the deepest non glob ancestor of table.getSc().getLocation() iff getLocation() points to
+   *                           a single glob. (e.g. /path/to/*&#47;files -> /path/to). If getLocation() contains none
+   *                           or multiple globs, job will fail.
+   *     relative_path := path of the file relative to origin_table_root. If the path of the file is not a descendant
+   *                      of origin_table_root, job will fail.
+   *     target_path := /resolved/table/root/relative/path
+   *     This mode is useful when moving a table with a complicated directory structure to a different base directory.
+   *   * if {@link #COPY_TARGET_TABLE_ROOT} is not defined in the properties, then the target is identical to the origin
+   *     path.
+   * </p>
+   */
+  @Override public Collection<CopyableFile> getCopyableFiles(FileSystem targetFs, CopyConfiguration configuration)
+      throws IOException {
+
+    List<CopyableFile> copyableFiles = Lists.newArrayList();
+
+    try(AutoReturnableObject<IMetaStoreClient> client = new AutoReturnableObject<>(this.clientPool)) {
+      if (isPartitioned()) {
+        List<Partition> partitions = client.get().listPartitions(this.table.getDbName(), this.table.getTableName(),
+            Short.MAX_VALUE);
+        for (Partition partition : partitions) {
+
+          StorageDescriptor storage = partition.getSd();
+
+          for (CopyableFile.Builder builder : getCopyableFilesFromStorageDescriptor(storage, configuration,
+              Optional.of(partition))) {
+            copyableFiles.add(builder.fileSet(gson.toJson(partition.getValues())).build());
+          }
+
+        }
+      } else {
+        for (CopyableFile.Builder builder : getCopyableFilesFromStorageDescriptor(this.table.getSd(), configuration,
+            Optional.<Partition>absent())) {
+          copyableFiles.add(builder.build());
+        }
+      }
+    } catch (TException te) {
+      throw new IOException(te);
+    }
+    return copyableFiles;
+  }
+
+  /**
+   * Get builders for a {@link CopyableFile} for each file referred to by a {@link StorageDescriptor}.
+   */
+  private List<CopyableFile.Builder> getCopyableFilesFromStorageDescriptor(StorageDescriptor storageDescriptor,
+      CopyConfiguration configuration, Optional<Partition> partition)
+      throws IOException {
+    List<CopyableFile.Builder> builders = Lists.newArrayList();
+    List<SourceAndDestination> dataFiles = Lists.newArrayList();
+
+    for (Path path : parseLocationIntoPaths(storageDescriptor.getLocation())) {
+
+      FileStatus[] statuses = this.fs.globStatus(path);
```

Comment:
I don't think so, AFAIK if the glob matches a hidden file, Hive will still match it.